pkg/assets/static/redirectpage.js

```diff
@@ -0,0 +1,189 @@
+}
+
+function handleError(error) {
+  window.location.href = consoleUrlMock;
+}
+
+function handleUnauthenticated() {
+  idToken = null;
+  window.location.href = consoleUrlMock;
+}
+
+function handleUserInfo() {
+  idToken = keycloak.idToken;
+}
+
+function refreshToken() {
+  keycloak.updateToken(30).catch((error) => {
+    console.error("Failed to refresh token:", error);
+    handleUnauthenticated();
+  });
+}
+
+function initializeKeycloak(clientConfig) {
+  keycloak = new Keycloak(clientConfig);
+  keycloak
+    .init({
+      onLoad: "check-sso",
+      silentCheckSsoRedirectUri:
+        window.location.origin + "/silent-check-sso.html",
+    })
+    .then((authenticated) => {
+      if (authenticated) {
+        console.log("User is authenticated");
+        intervalRefRefresh = setInterval(refreshToken, 15000); //start 15s interval token refresh
+        keycloak
+          .loadUserInfo()
+          .then(handleUserInfo)
+          .catch(() => handleError("failed to pull in user data."));
+      } else {
+        console.log("user not authenticated");
+        handleUnauthenticated();
+      }
+    })
+    .catch(() => handleError("Failed to initialize authorization"));
+}
+
+// General function to fetch JSON data
+function getJSON(
+  method,
+  url,
+  token,
+  callback,
+  body = null,
+  headers = {}
+) {
+  let xhr = new XMLHttpRequest();
+  xhr.open(method, url, true);
+  if (token != null) xhr.setRequestHeader("Authorization", "Bearer " + token);
+
+  Object.entries(headers).forEach(([key, value]) => {
+    xhr.setRequestHeader(key, value);
+  });
+
+  xhr.responseType = "json";
+  xhr.onload = () => {
+    let status = xhr.status;
+    if (status >= 200 && status < 300) {
+      callback(null, xhr.response);
+    } else {
+      callback(status, xhr.response);
+    }
+  };
+  xhr.send(body ? JSON.stringify(body) : null);
+}
+
+function loadAuthLibrary(url, cbSuccess, cbError) {
+  const script = document.createElement("script");
+  script.setAttribute("src", url);
+  script.setAttribute("type", "text/javascript");
+  let loaded = false;
+  function loadFunction() {
+    if (loaded) return;
+    loaded = true;
+    cbSuccess();
+  }
+  function errorFunction(error) {
+    if (loaded) return;
+    cbError(error);
+  }
+  script.onerror = errorFunction;
+  script.onload = loadFunction;
+  script.onreadystatechange = loadFunction;
+  document.head.appendChild(script);
+}
+
+async function getRedirectData() {
+  const xhr = new XMLHttpRequest();
+
+  xhr.open("GET", registrationURL, true);
+
+  xhr.setRequestHeader("Authorization", `Bearer ${idToken}`);
+  xhr.onreadystatechange = function () {
+    if (xhr.readyState === XMLHttpRequest.DONE) {
+      if (xhr.status >= 200 && xhr.status < 300) {
+        try {
+          const data = JSON.parse(xhr.responseText);
+          handleSuccess(data);
+        } catch (error) {
+          handleError();
+        }
+      } else {
+        handleError();
```

Comment:
If the signup api returns 404 it's not an error actually. It just means that user has to sign up first. So I wouldn't mix it with other cases when something went wrong.
And yes, in case of 404 we should redirect to the signup URL but we also need to redirect back to the redirectpage after user signs up successfully. See https://issues.redhat.com/browse/RHCLOUD-32115?focusedId=24554951&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-24554951 for details.
The redirect param is not currently supported in HCC but we can add this to our internal landing page for now and later add it to HCC too.